### PR TITLE
Update kn tutorials for 0.9.2

### PR DESCRIPTION
**Pull request by jonnyzzz**
Update tutorial to include missing features

### CODE REVIEW

File: pages/docs/tutorials/native/apple-framework.md [hunk 1]
```diff
@@ -138,6 +142,44 @@ Most of the collection types are mapped to similar collection types from the oth
...
+
+```obj-c
+- (instancetype)initWith__TYPE_:(__TYPE__)value __attribute__((unavailable));
++ (instancetype)numberWith__TYPE_:(__TYPE__)value __attribute__((unavailable));
+```
+
+</div>
+Where `__TYPE__` is one of the simple type names, capitalized if necessary, e.g. `initWithChar(char)`.
+
+Those types are used to map boxed Kotlin number types into Objective-C and Swift.
```
Comment (B1ggDave) [inline pages/docs/tutorials/native/apple-framework.md]: These types...

File: pages/docs/tutorials/native/dynamic-libraries.md [hunk 2]
```diff
@@ -232,26 +248,29 @@ documentation or to check the related tutorial named [Kotlin/Native as an Apple
 
 ### Our Library Functions
 
-Let's take a look on the `kotlin.root.demo` field, it
+Let's take a look on the `kotlin.root.example` field, it
```
Comment (B1ggDave) [inline pages/docs/tutorials/native/dynamic-libraries.md]: Let's take a look at the 'kotlin.root.example'

File: pages/docs/tutorials/native/mapping-primitive-data-types-from-c.md [hunk 3]
```diff
@@ -44,6 +44,7 @@ We create a `lib.h` file to see how C functions are mapped into Kotlin:
 #define LIB2_H_INCLUDED
 
 void ints(char c, short d, int e, long f);
+void uints(unsigned char c, unsigned short d, unsigned int e, unsigned long f);
```
Comment (B1ggDave) [inline pages/docs/tutorials/native/mapping-primitive-data-types-from-c.md]: is 'uints' right?

Comment (jonnyzzz) [reply pages/docs/tutorials/native/mapping-primitive-data-types-from-c.md]: that is correct, it is the name of a function

File: pages/docs/tutorials/native/apple-framework.md [hunk 4]
```diff
@@ -138,6 +142,44 @@ Most of the collection types are mapped to similar collection types from the oth
...
+-------|-----|-----------|
+|`-`      |`KotlinNumber` |`<Package>Number` | `-` |
+|`Byte`   |`KotlinByte`   |`<Package>Byte`   | `char` |
+|`UByte`  |`KotlinUByte`  |`<Package>UByte`  | `unsigned char` |
+|`Short`  |`KotlinShort`  |`<Package>Short`  | `short` |
+|`UShort` |`KotlinUShort` |`<Package>UShort` | `unsigned short` |
+|`Int`    |`KotlinInt`    |`<Package>Int`    | `int` |
+|`UInt`   |`KotlinUInt`   |`<Package>UInt`   | `unsigned int` |
+|`Long`   |`KotlinLong`   |`<Package>Long`   | `long` |
+|`ULong`  |`KotlinULong`  |`<Package>ULong`  | `unsigned long` |
```
Comment (SvyatoslavScherbina) [inline pages/docs/tutorials/native/apple-framework.md]: Kotlin `(U)Long` corresponds to `(unsigned) long long` C type.

File: pages/docs/tutorials/native/apple-framework.md [hunk 5]
```diff
@@ -138,6 +142,44 @@ Most of the collection types are mapped to similar collection types from the oth
...
+{:.wide.zebra}
+
+Every number type has a class method to create a new instance from a simple type. Also, there are all instance methods
+to extract a simple value back. Schematically, declarations look like that,
+
+<div clacss="sample" markdown="1" mode="obj-c" theme="idea" data-highlight-only auto-indent="false">
+
+```obj-c
+- (instancetype)initWith__TYPE_:(__TYPE__)value __attribute__((unavailable));
++ (instancetype)numberWith__TYPE_:(__TYPE__)value __attribute__((unavailable));
```
Comment (SvyatoslavScherbina) [inline pages/docs/tutorials/native/apple-framework.md]: ` __attribute__((unavailable))` is used to mask constructors inherited from `NSNumber`, so the only allowed constructors are those without this attribute.

File: pages/docs/tutorials/native/mapping-primitive-data-types-from-c.md [hunk 6]
```diff
@@ -108,16 +110,17 @@ as it is usually an 8-bit unsigned value.
 | C | Kotlin |
 |---|--------|
 | char  |  kotlin.Byte |
+| unsigned char  |  kotlin.UByte |
 | short |  kotlin.Short |
+| unsigned short |  kotlin.UShort |
 | int   |  kotlin.Int |
+| unsigned int   |  kotlin.UInt |
 | long  |  kotlin.Long |
+| unsigned long  |  kotlin.ULong |
```
Comment (SvyatoslavScherbina) [inline pages/docs/tutorials/native/mapping-primitive-data-types-from-c.md]: C `long` type can be either 32-bit or 64-bit depending on the platform, so it is mapped to Kotlin `Int` or `Long`.
However `long long` type is (usually) mapped to Kotlin `Long`.

File: pages/docs/tutorials/native/apple-framework.md (hunk 4 above)
Comment (jonnyzzz) [reply pages/docs/tutorials/native/apple-framework.md]: Fixed. Where do we map 
```
- (instancetype)initWithLong:(long)value __attribute__((unavailable));
- (instancetype)initWithUnsignedLong:(unsigned long)value __attribute__((unavailable));
```

File: pages/docs/tutorials/native/apple-framework.md (hunk 5 above)
Comment (jonnyzzz) [reply pages/docs/tutorials/native/apple-framework.md]: fixed

File: pages/docs/tutorials/native/apple-framework.md [hunk 7]
```diff
@@ -138,6 +142,44 @@ Most of the collection types are mapped to similar collection types from the oth
...
+```
+
+</div>
+Where `__TYPE__` is one of the simple type names, capitalized if necessary, e.g. `initWithChar(char)`.
```
[the remark this answers is not in the document]
Comment (jonnyzzz) [reply pages/docs/tutorials/native/apple-framework.md]: fixed

File: pages/docs/tutorials/native/apple-framework.md [hunk 8]
```diff
@@ -178,12 +223,12 @@ respectively. The `Interface` is turned into `@protocol`, both a `class` and an
...
 The `Demo` prefix comes from the `-output` parameter
 of the `kotlinc-native` compiler and the framework name. 
-You may have spotted that the nullable return type `Long?` was turned into `NSNumber*` in Objective-C.
+You may have spotted that the nullable return type `ULong?` was turned into `DemoLong*` in Objective-C.
 
 ### Global Declarations from Kotlin
 
 All global functions from Kotlin
-are turned into `Demo` class in Objective-C/Swift, where `Demo` is the framework name and set by
+are turned into `DemoLibKt` class in Objective-C/Swift, where `Demo` is the framework name and set by
```
[the remark this answers is not in the document]
Comment (jonnyzzz) [reply pages/docs/tutorials/native/apple-framework.md]: fixed. Thanks

File: pages/docs/tutorials/native/mapping-primitive-data-types-from-c.md (hunk 6 above)
Comment (jonnyzzz) [reply pages/docs/tutorials/native/mapping-primitive-data-types-from-c.md]: Got it. Thanks. Fixed it as well

File: pages/docs/tutorials/native/apple-framework.md (hunk 4 above)
Comment (SvyatoslavScherbina) [reply pages/docs/tutorials/native/apple-framework.md]: We do not map these ones as Kotlin doesn't have a concept like C `long` type.
The same is true for `initWith(Unsigned)Integer` with `NS(U)Integer` types.

**Pull request merged**